clang/test/SemaSYCL/sycl-pfwg-invalid-code.cpp

```diff
@@ -0,0 +1,14 @@
+// RUN: %clang_cc1 -fsycl-is-device %s -verify
+
+// Tests that an error diagnostic is issued without a crash (due to a
+// triggered assertion) when some invalid cases are encountered in
+// the processing of kernel_parallel_for_work_group.
+
+template <typename, typename, typename K>
```

Comment:
The crash happened earlier when incorrect code was used in library? I thought it was invalid code in user code.

Reply:
In the original example, the user code is invalid in such a way that when the call gets to the library, the invocation within parallel_for_work_group fails after a few cascading template instantiation failures.  This is hard (for me) to reproduce with the mock headers even after bringing a lot of code in.  Hence the modified example.

Reply:
Ok. Thanks for clarifying